pkg/lib/cockpit.d.ts

```diff
@@ -193,13 +193,31 @@ declare module 'cockpit' {
 
     /* === String helpers ======================== */
 
+    type FormatOptions = {
+        precision?: number;
+        separate?: boolean;
+    };
+
+    type ByteUnit = {
+        name: string | null;
+        factor: number;
+        selected?: boolean;
+    };
+
     function message(problem: string | JsonObject): string;
 
     function gettext(message: string): string;
     function gettext(context: string, message?: string): string;
     function ngettext(message1: string, messageN: string, n: number): string;
     function ngettext(context: string, message1: string, messageN: string, n: number): string;
 
-    function format_bytes(n: number): string;
     function format(format_string: string, ...args: unknown[]): string;
+    function format_number(n: number, precision?: number): string
+    function format_bytes(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate?: false }): string;
+    function format_bytes(n: number, factor: 1000 | 1024, options: FormatOptions & { separate: true }): string[];
+    function format_bytes_per_sec(n: number, factor?: 1000 | 1024, options?: FormatOptions & { separate?: false }): string;
+    function format_bytes_per_sec(n: number, factor: 1000 | 1024, options: FormatOptions & { separate: true }): string[];
+    function format_bits_per_sec(n: number, factor?: 1000 | 1024, options?: FormatOptions & { seperate?: false }): string;
```

Comment:
Sorry I didn't catch this on an earlier round, but here you spell "separate" as "seperate"...

Reply:
That's something I commonly typo :(